develop/tutorials/articles/110-portlets/15-javascript/04-npm/02-adding-liferay-npm-bundler-to-your-portlet.markdown

```diff
@@ -0,0 +1,63 @@
+# Adding liferay-npm-bundler to Your Portlet [](id=adding-liferay-npm-bundler-to-your-portlet)
+
+Adding liferay-npm-bundler to your portlet involves installing the package, via 
+npm, and adding it to your npm build process. Get started by installing 
+liferay-npm-bundler next.
+
+## Installing liferay-npm-bundler [](id=installing-liferay-npm-bundler)
+
+Follow these steps to install liferay-npm-bundler:
+
+1.  Install NodeJS >= [v6.11.0](http://nodejs.org/dist/v6.11.0/) if you 
+    don't have it installed.
+
+2.  Navigate to your portlet's `src/main/resources/META-INF/resources` 
+    folder.
+     
+    If you don't have a portlet already, create an empty MVC portlet. For 
+    convenience, you can use [Blade CLI](https://dev.liferay.com/develop/tutorials/-/knowledge_base/7-0/installing-blade-cli) 
+    to create an empty portlet with the [mvc portlet blade template](https://dev.liferay.com/develop/reference/-/knowledge_base/7-0/using-the-mvc-portlet-template).
+
+3.  Run the following command to install the liferay-npm-bundler:
+
+        npm install --save-dev liferay-npm-bundler
+
+    +$$$
+
+    **Note:** Use npm from within your portlet's 
+    `src/main/resources/META-INF/resources` folder, as you normally do on a 
+    typical web project.
+
+    $$$
+
+Now that you have the liferay-npm-bundler installed, you can add it to your 
+npm build process next.
+
+## Adding liferay-npm-bundler to Your Build Process [](id=adding-liferay-npm-bundler-to-your-build-process)
+
+After everything is transpiled (if necessary) to ECMAScript in your build 
```

Comment:
After 'everything is transpiled (if necessary) to ECMAScript' I would add 'and converted to AMD format'.

Reply:
I mean, it is the responsibility of the developer (not of the bundler) to transform his own code and produce a portal-suitable Javascript.

Then bundler only takes care of code inside npm packages, but not of the code of the project.

Reply:
Is this better?
After any JavaScript in your file is transpiled (if necessary) to ECMAScript in 
your build script and converted to AMD format, you must run liferay-npm-bundler 
in your `package.json` to pack the needed npm packages and transform them to AMD. 
This lets [Liferay AMD Loader](https://github.com/liferay/liferay-amd-loader) 
grab them from the Portal.

Reply:
After any source code in your file is transpiled (if necessary) to ECMAScript in 
your build script and converted to AMD format for the [Liferay AMD Loader](/develop/tutorials/-/knowledge_base/7-0/liferay-amd-module-loader), you 
must run liferay-npm-bundler in your `package.json` to pack the needed npm 
packages and transform them to AMD. This lets Liferay AMD Loader grab the 
packages from the Portal.